src/components/PoolInfo.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect } from 'react'
+import { Box } from '@mui/system'
+import { Typography } from '@mui/material'
+import { fromDecimals } from 'utils'
+import { ton } from 'services/api/addresses'
+import { client } from 'services/api'
+import useUsdValue from 'hooks/useUsdValue'
+import { useTokenOperationsStore } from 'store/token-operations/hooks'
+import { useTranslation } from 'react-i18next'
+import { usePoolInfo } from 'store/pool-info/hooks'
+
+const calculateDecimals = (val: string) => {
+  const n = parseFloat(val)
+  if (n > 1) return n.toFixed(2)
+  else if (n < 0.01) return n.toFixed(6)
+  else return n.toFixed(4)
+}
+
+export const PoolInfo = () => {
+  const {fetchPoolData, poolInfo} = usePoolInfo()
+  const {usd} = useUsdValue(ton.name, fromDecimals(poolInfo.tonReserves?.muln(2) || 0, ton.decimals))
+  const { selectedToken } = useTokenOperationsStore();
+  const {t} = useTranslation()
+
```

Comment:
Please make sure to clear the pool info inside the reducer when component unmount.
because if we dont clear it, so the next time he will enter the page with different token details, he will see the previous poolinfo data.